src/EnergyPlus/Data/EnergyPlusData.hh

```diff
@@ -0,0 +1,195 @@
+//     conditions and the following disclaimer in the documentation and/or other materials
+//     provided with the distribution.
+//
+// (3) Neither the name of the University of California, Lawrence Berkeley National Laboratory,
+//     the University of Illinois, U.S. Dept. of Energy nor the names of its contributors may be
+//     used to endorse or promote products derived from this software without specific prior
+//     written permission.
+//
+// (4) Use of EnergyPlus(TM) Name. If Licensee (i) distributes the software in stand-alone form
+//     without changes from the version obtained under this License, or (ii) Licensee makes a
+//     reference solely to the software portion of its product, Licensee must refer to the
+//     software as "EnergyPlus version X" software, where "X" is the version number Licensee
+//     obtained under this License and may not use a different name for the software. Except as
+//     specifically required in this Section (4), Licensee shall not use in a company name, a
+//     product name, in advertising, publicity, or other promotional activities any name, trade
+//     name, trademark, logo, or other designation of "EnergyPlus", "E+", "e+" or confusingly
+//     similar designation, without the U.S. Department of Energy's prior written consent.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#ifndef Globals_hh_INCLUDED
+#define Globals_hh_INCLUDED
+
+// EnergyPlus Headers
+#include <EnergyPlus/EnergyPlus.hh>
+
+#include <string>
+
+struct BaseGlobalStruct
+{
+    virtual void clear_state() = 0;
+};
+
+//struct OutputReportTabular : BaseGlobalStruct
+//{
+//    //int MaxHeaderLength;
+//
+//    void clear_state() override {
+//    }
+//};
+
+struct DataGlobal : BaseGlobalStruct
+{
+    // Data
+    bool AnnualSimulation = false;
+
+    // MODULE VARIABLE DECLARATIONS:
+    std::string DayOfSimChr = "0";       // Counter for days (during the simulation) (character -- for reporting)
+
+    // MODULE PARAMETER DEFINITIONS
+    static constexpr int EndZoneSizingCalc = 4;
+
+    void clear_state() override {
+        AnnualSimulation = false;
+        DayOfSimChr = "0";
+    }
+};
+
+struct ExteriorEnergyUseGlobals : BaseGlobalStruct
+{
+    struct ExteriorLightUsage

```

Comment:
moved this struct to here from https://github.com/NREL/EnergyPlus/blob/global/src/EnergyPlus/ExteriorEnergyUse.hh#L101-L123